### ISSUE TITLE
Problem running preprocessing for flowers retrain tutorial

### ISSUE THREAD

**Issue by StephenBydawell** [GoogleCloudPlatform/cloudml-samples#205]
I'm doing the 'Image Classification using Flowers dataset' tutorial on cloud shell: https://cloud.google.com/ml-engine/docs/tensorflow/flowers-tutorial

when running: 
```
python trainer/preprocess.py --input_dict "gs://cloud-ml-data/img/flower_photos/dict.txt" --input_path "gs://cloud-ml-data/img/flower_photos/eval_set.csv" --output_path "gs://retrain-flowers-mlengine/stephen/flowers_stephen_20180607_143118/preproc/eval" --cloud
```
I get the following:

```
/usr/local/lib/python2.7/dist-packages/apache_beam/coders/typecoders.py:132: UserWarning: Using fallback coder for typehint: Any.
  warnings.warn('Using fallback coder for typehint: %r.' % typehint)

Usage:
  /usr/bin/python -m pip install [options] <requirement specifier> [package-index-options] ...
  /usr/bin/python -m pip install [options] -r <requirements file> [package-index-options] ...
  /usr/bin/python -m pip install [options] [-e] <vcs project url> ...
  /usr/bin/python -m pip install [options] [-e] <local project path> ...
  /usr/bin/python -m pip install [options] <archive url/path> ...

no such option: --download
Traceback (most recent call last):
  File "trainer/preprocess.py", line 490, in <module>
    main(sys.argv[1:])
  File "trainer/preprocess.py", line 486, in main
    run(arg_dict)
  File "trainer/preprocess.py", line 394, in run
    configure_pipeline(p, in_args)
  File "/usr/local/lib/python2.7/dist-packages/apache_beam/pipeline.py", line 170, in __exit__
    self.run().wait_until_finish()
  File "/usr/local/lib/python2.7/dist-packages/apache_beam/pipeline.py", line 163, in run
    return self.runner.run(self)
  File "/usr/local/lib/python2.7/dist-packages/apache_beam/runners/dataflow/dataflow_runner.py", line 175, in run
    self.dataflow_client.create_job(self.job), self)
  File "/usr/local/lib/python2.7/dist-packages/apache_beam/utils/retry.py", line 174, in wrapper
    return fun(*args, **kwargs)
  File "/usr/local/lib/python2.7/dist-packages/apache_beam/runners/dataflow/internal/apiclient.py", line 411, in create_job
    self.create_job_description(job)
  File "/usr/local/lib/python2.7/dist-packages/apache_beam/runners/dataflow/internal/apiclient.py", line 432, in create_job_description
    job.options, file_copy=self._gcs_file_copy)
  File "/usr/local/lib/python2.7/dist-packages/apache_beam/runners/dataflow/internal/dependency.py", line 362, in stage_job_resources
    _stage_beam_sdk_tarball(sdk_remote_location, staged_path, temp_dir)
  File "/usr/local/lib/python2.7/dist-packages/apache_beam/runners/dataflow/internal/dependency.py", line 447, in _stage_beam_sdk_tarball
    _dependency_file_copy(_download_pypi_sdk_package(temp_dir), staged_path)
  File "/usr/local/lib/python2.7/dist-packages/apache_beam/runners/dataflow/internal/dependency.py", line 510, in _download_pypi_sdk_package
    processes.check_call(cmd_args)
  File "/usr/local/lib/python2.7/dist-packages/apache_beam/utils/processes.py", line 40, in check_call
    return subprocess.check_call(*args, **kwargs)
  File "/usr/lib/python2.7/subprocess.py", line 186, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/usr/bin/python', '-m', 'pip', 'install', '--download', '/tmp/tmpD_dkHj', 'apache-beam==0.6.0', '--no-binary', ':all:', '--no-deps']' returned non-zero exit status 2
```
I have completed all the previous steps using cloud shell.

**Comment by StephenBydawell** [comment]
Updating apache-beam has seemed to work:
```
pip installed apache-beam --upgrade
```

**Comment by happyhuman** [comment]
Thanks for posting the solution. I am going to go ahead and close this issue.

**Comment by fatemeh91** [comment]
I have the same Issue but upgrading the apache-beam even does not work

**Comment by LefterisS** [comment]
The problem is with the `--download` flag in [this](https://github.com/apache/beam/blob/release-0.6.0/sdks/python/apache_beam/runners/dataflow/internal/dependency.py#L506) line. One solution is to use an older version of pip (<=9.0.3) where the flag is not yet deprecated. Even better though would be if the requirements.txt included a newer version of beam (>=2.3.0) where "pip install --download" is no longer used.

**Comment by spicyramen** [comment]
I upgraded to apache-beam==2.6.0 and this seems to have fixed the issue. I agree requirements.txt needs to be updated.
